### PR TITLE
AVRO-3748 [java] fix SeekableInputStream.skip

### PR DESCRIPTION
**Pull request by steven-aerts**
The implementation of SeekableInputStream.skip had a longstantind issue that it did a seek to a wrong position.
This issue is rather difficult to hit in avro as it is only used in corner cases of for example the FastReader.

## What is the purpose of the change

* Fixes AVRO-3748 a longstanding issue in SeekableInputStream.skip

## Verifying this change

This change added a test to verify the new behavior 

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)

### CODE REVIEW

File: lang/java/avro/src/main/java/org/apache/avro/file/DataFileReader.java [hunk 1]
```diff
@@ -312,10 +312,10 @@ public long skip(long skip) throws IOException {
       long length = in.length();
       long remaining = length - position;
       if (remaining > skip) {
-        in.seek(skip);
+        in.seek(position + skip);
         return in.tell() - position;
```
Comment (clesaec) [inline lang/java/avro/src/main/java/org/apache/avro/file/DataFileReader.java:316]: And in same time, either `return skip; ... else { ... return length - position; }`  either put `return in.tell() - position;` at the end of function.

**Review by clesaec — APPROVED** [review]
LGTM

File: lang/java/avro/src/test/java/org/apache/avro/file/TestSeekableInputStream.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+public class TestSeekableInputStream {
+    @Test
+    public void testSkip(@TempDir Path tempDir) throws IOException {
+        File f = tempDir.resolve("testSkip.avro").toFile();
+        try (FileWriter w = new FileWriter(f)) {
+            w.write("someContent");
+        }
+        try(DataFileReader.SeekableInputStream stream =
+                    new DataFileReader.SeekableInputStream(new SeekableFileInput(f))
```
[the remark this answers is not in the document]
Comment (steven-aerts) [reply lang/java/avro/src/test/java/org/apache/avro/file/TestSeekableInputStream.java]: Done

File: lang/java/avro/src/test/java/org/apache/avro/file/TestSeekableInputStream.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        ) {
+            for (int i = 0; i < stream.length(); i++) {
```
[the remark this answers is not in the document]
Comment (steven-aerts) [reply lang/java/avro/src/test/java/org/apache/avro/file/TestSeekableInputStream.java]: Done
